HW7(Tests)/Functions.js

```diff
@@ -0,0 +1,23 @@
+const { getUsers } = require('./Async-functions');
+
+function validateEmail(email) {
+  return /^\s*\w{5,}@[a-z]+\.[a-z]+\s*$/i.test(email);
+}
+
+class User {
+  constructor (firstName = 'Homer', lastName = 'Simpson') {
+    this.firstName = firstName;
+    this.lastName = lastName;
+  }
+
+  getFullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+}
+
+async function getUsersEmails() {
+  const users = await getUsers();
+  return users.map(user => user.email);
+}
```

Comment:
it is better to put the logic for receiving an email into a separate clean function.
it will be easier to maintain and test.

and sequentially call a function to obtain general information on users and a function to transform / filter this data.